src/main/java/seedu/address/model/time/TimeMask.java

```diff
@@ -72,15 +77,42 @@ public void mergeIsolatedEvents(IsolatedEventList isolatedEventList, LocalDate s
                 endTime = LAST_HOUR_INDEX;
             }
 
-            if (startIndex != endIndex) {
-                occupySlots(startIndex, startTime, endTime);
-                occupySlots(endIndex, 0, endTime);
+            if (!start.isEqual(end)) {
+                // check if end date after date limit
+                boolean isAfterDateLimit = isEndDateAfterDateLimit(end.toLocalDate(), dateLimit);
+                // get the days between the start and end/ start and date limit
+                long daysBetween = getDaysBetween(start.toLocalDate(), end.toLocalDate(), dateLimit);
+                // if end date after date limit, need to change end time of event
+                int modifiedEndTime = getUpdatedEndTime(endTime, isAfterDateLimit);
+                occupyMultipleDay(startIndex, daysBetween, startTime, modifiedEndTime);
             } else {
                 occupySlots(startIndex, startTime, endTime);
             }
         }
     }
 
+    private int getUpdatedEndTime(int endTime, boolean isAfterDateLimit) {
+        if (isAfterDateLimit) {
+            return 23;
```

Comment:
Should make use of the existing `LAST_HOUR_INDEX` instead of using 'magic numbers'